### PR TITLE
Glue templates

### PR DESCRIPTION
**Pull request by ibrand**
Glue job templating done such that the glue job pulls the latest signatures_job.py script from csl when we terraform deploy so we don't need to manually update the glue code anymore

### CODE REVIEW

File: glue_job.tf [hunk 1]
```diff
@@ -4,20 +4,14 @@ resource "aws_glue_catalog_database" "catalog_db" {
...
   signatures_s3_path = "s3://agra-data-exports-${var.controlshift_environment}/${var.controlshift_organization_slug}/full/signatures"
+  signatures_table_index = index(local.parsed_bulk_data_schemas.tables.*.table.name, "signatures")
+  signatures_table_columns = local.parsed_bulk_data_schemas.tables[local.signatures_table_index].table.columns
 }
 
 resource "aws_glue_crawler" "signatures_crawler" {
   database_name = aws_glue_catalog_database.catalog_db.name
   name = "${var.controlshift_environment}_full_signatures"
   role = aws_iam_role.glue_service_role.arn
-  configuration = jsonencode(
```
Comment (sjwmoveon) [inline glue_job.tf]: Why are we getting rid of this bit?

File: glue_job.tf [hunk 2]
```diff
@@ -26,45 +20,60 @@ resource "aws_glue_crawler" "signatures_crawler" {
...
-        sse_algorithm = "AES256"
-      }
-    }
-  }
+resource "aws_s3_bucket_acl" "glue_resources" {
+  bucket = aws_s3_bucket.glue_resources.id
+  acl    = "private"
+}
 
-  lifecycle_rule {
```
Comment (sjwmoveon) [inline glue_job.tf:39]: Seems like we would want to keep this rule for deleting temp files?

**Review by sjwmoveon — CHANGES_REQUESTED** [review]
This mostly makes sense but I have a couple questions.

**Comment by ibrand** [comment]
So, this was a direct copypaste from https://github.com/controlshift/terraform-aws-controlshift-redshift-sync/blob/main/glue_job.tf which is why some of those blocks were removed. I think either one or both of those blocks might need to be written differently in the newer terraform format (for example pasting the `lifecycle_rule` block back in throws errors) -- do you think it's worth debugging?

File: glue_job.tf (hunk 1 above)
Comment (sjwmoveon) [reply glue_job.tf]: https://github.com/MoveOnOrg/terraform-aws-controlshift-redshift-sync/commit/4a2e1c32a1036e892a3e6fb647c2b7d51fd34b9c looks like we manually set this part of the configuration so we may want to keep it.

File: glue_job.tf (hunk 2 above)
Comment (sjwmoveon) [reply glue_job.tf:39]: Similarly we added this manually added lifecycle rule to the config: https://github.com/MoveOnOrg/terraform-aws-controlshift-redshift-sync/commit/19be449c3420f05b3f590e2baa8611d59472ae67 and https://github.com/MoveOnOrg/terraform-aws-controlshift-redshift-sync/commit/7fa1de1bf76bd0739438572e8cfcea5c1b6b4113

**Comment by sjwmoveon** [comment]
I think it's worth debugging since both changes were made intentionally on our end to do some extra cleanup.

File: glue_job.tf [hunk 3]
```diff
@@ -153,8 +162,6 @@ resource "aws_glue_job" "signatures_full" {
   name = "cs-${var.controlshift_environment}-signatures-full"
   connections = [ aws_glue_connection.redshift_connection.name ]
   glue_version = "3.0"
-  number_of_workers = 9
```
Comment (sjwmoveon) [inline glue_job.tf]: This also was set manually

**Comment by ibrand** [comment]
Oh I see that you added the manual changes back in! Does this mean that you got it to deploy without errors with them?

**Comment by sjwmoveon** [comment]
I did, yes!  No luck with the results of the Glue job, though.

**Comment by ibrand** [comment]
I'd say we're good to merge in then!

**Review by sjwmoveon — APPROVED** [review]
Approved with the more recent changes.  @ibrand you may want to take another look as well since I made the last few commits.

**Review by sjwmoveon — CHANGES_REQUESTED** [review]
I think we want to undo all of this since we're not going with the templated Glue job any more.  Fortunately it was never checked in.

**Comment by ibrand** [comment]
Great! I didn't realize it was never checked in. Closing this.

**Pull request closed without merging**